FP_analysis_of_reviews_from_Amazon/test_nearest_review.py

```diff
@@ -0,0 +1,327 @@
+            ],
+            "3": [ReviewTimeInfo(200, None)],
+        }
+        bots = {"1", None}
+        analyzed_data, number_of_bot_comments = filter_bot_review(data, bots)
+        self.assertEqual(number_of_bot_comments, 2)
+        self.assertEqual(
+            analyzed_data,
+            {
+                "2": [
+                    ReviewTimeInfo(time=100, text=None),
+                    ReviewTimeInfo(time=300, text="c"),
+                    ReviewTimeInfo(time=400, text="d"),
+                ],
+                "3": [ReviewTimeInfo(time=200, text=None),],
+            },
+        )
+        self.assertIsInstance(analyzed_data, dict)
+        self.assertTrue(
+            all(
+                isinstance(i.text, str) or i.text is None
+                for v in analyzed_data.values()
+                for i in v
+            )
+        )
+    def test_filter_bot_review_without_bots(self):
+        data = {
+            "1": [ReviewTimeInfo(100, "a"), ReviewTimeInfo(200, "b")],
+            "2": [ReviewTimeInfo(100, None)],
+        }
+        analyzed_data, number_of_bot_comments = filter_bot_review(data, set())
+        self.assertEqual(number_of_bot_comments, 0)
+        self.assertEqual(
+            analyzed_data,
+            {
+                "1": [
+                    ReviewTimeInfo(time=100, text="a"),
+                    ReviewTimeInfo(time=200, text="b"),
+                ],
+                "2": [ReviewTimeInfo(time=100, text=None)],
+            },
+        )
+
+        analyzed_data, number_of_bot_comments = filter_bot_review(data, {None})
+        self.assertEqual(number_of_bot_comments, 0)
+        self.assertEqual(
+            analyzed_data,
+            {
+                "1": [
+                    ReviewTimeInfo(time=100, text="a"),
+                    ReviewTimeInfo(time=200, text="b"),
+                ],
+                "2": [ReviewTimeInfo(time=100, text=None)],
+            },
+        )
+    def test_filter_bot_review_when_all_bots(self):
+        data = {
+            "1": [ReviewTimeInfo(100, "a"), ReviewTimeInfo(100, "b")],
+            "2": [ReviewTimeInfo(100, None), ReviewTimeInfo(100, None)],
+        }
+        bots = {"1", "2", None}
+        analyzed_data, number_of_bot_comments = filter_bot_review(data, bots)
+        self.assertEqual(number_of_bot_comments, 4)
+        self.assertEqual(analyzed_data, {})
+
+class TestProcessData(TestCase):
+    def test_process_data_with_bots(self):
+        data = {
+            "1": [ReviewTimeInfo(100, "a"), ReviewTimeInfo(100, "b")],
+            "2": [
+                ReviewTimeInfo(100, None),
+                ReviewTimeInfo(300, "c"),
+                ReviewTimeInfo(400, "d"),
+            ],
+            "3": [ReviewTimeInfo(200, None)],
+        }
+        analyzed_data, potential_bots, number_of_bot_comments = process_data(data)
+        self.assertTrue(v is ReviewTimeInfo for v in analyzed_data.values())
```

Comment:
I think you've accidentally missed this one

Reply:
Yes. I deleted it now